arbor/communication/communicator.cpp

```diff
@@ -68,16 +79,25 @@ void communicator::update_connections(const connectivity& rec,
     for (auto g: dom_dec.groups()) {
         util::append(gids, g.gids);
     }
-    // Build the connection information for local cells in parallel.
-    std::vector<gid_info> gid_infos;
-    gid_infos.resize(num_local_cells_);
+
+    // For caching information about each cell
+    struct gid_info {
+        cell_gid_type gid;                          // global identifier of cell
+        cell_size_type index_on_domain;             // index of cell in this domain
+        std::vector<cell_connection> conns;         // list of connections terminating at this cell
+        std::vector<ext_cell_connection> ext_conns; // list of connections terminating at this cell from outside arbor
+    };
+
+    std::vector<gid_info> gid_infos(num_local_cells_, gid_info{});
     threading::parallel_for::apply(0, gids.size(), thread_pool_.get(),
         [&](cell_size_type i) {
             auto gid = gids[i];
-            gid_infos[i] = gid_info(gid, i, rec.connections_on(gid));
+            gid_infos[i] = {gid, i, rec.connections_on(gid), rec.external_connections_on(gid)};
         });
-    cell_local_size_type n_cons =
-        util::sum_by(gid_infos, [](const gid_info& g){ return g.conns.size(); });
+
+    cell_local_size_type
+        n_cons = util::sum_by(gid_infos, [](const auto& g){ return g.conns.size(); }),
+        n_ext_cons = util::sum_by(gid_infos, [](const auto& g){ return g.ext_conns.size(); });
```

Comment:
what's the harm in repeating the type? Please don't abuse the comma operator too much.

Reply:
Anything typed is too much. Also, I never considered this an operator use, for me this is just like `f(a, b, c)`.
If this can be overloaded I understand your aversion.